### DISCUSSION TITLE
`andRelatedTo` with a condition

### DISCUSSION THREAD

**Discussion by shifuma** [craftcms/cms#15886] [Q&A] [answered]
I'm looking to query for entries with relations like this:

```
{% set entries = craft.entries()
  .andRelatedTo({
    sourceElement: category,
    field: 'category',
    myFieldHandle: 'myFieldValue'
  })
  .all() %}
```

So basically, only get the relations where another field in the Matrix entry has a value of `myFieldValue`. I hope that makes sense.

Is that possible as is? If not, could something like that be made possible?

**Comment by shifuma** [comment]
Actually maybe this could be an extension of my other request: https://github.com/craftcms/cms/discussions/15790

If the opposite of that was added (for example a `nested` or `nestedId` param) I could instead do:

```
{% set nestedEntryIds = craft.entries().field('matrixField').myFieldHandle('myFieldValue').ids() %}
{% set entries = craft.entries().section('mySection').nestedId(nestedEntryIds).all() %}
```

**Comment by brandonkelly** [comment]
Can you clarify what you’re trying to do?

**Reply by shifuma** [reply]
I'm trying to filter entries based on what is inside a Matrix field. Specifically, based on the entry type.

```
Entry 1
- Matrix field has entries with entry type 1 & 2

Entry 2
- Matrix field has entries with entry type 2

Entry 3
- Matrix field has entries with entry type 2 & 3
```

So I want to query for all entries with entry type 1 & 3 in the Matrix field.

**Reply by brandonkelly** [reply, marked as answer]
Gotcha. The only current way to do that is by getting the IDs of the nested entries first, and then passing them to the Matrix field’s param on the parent entry query:

```twig
{% set nestedIds = craft.entries()
  .field('myMatrixField')
  .type(['type1Handle', 'type3Handle'])
  .ids() %}

{% if nestedIds|length %}
  {% set entries = craft.entries()
    .section('mySection')
    .myMatrixField(nestedIds)
    .all() %}
{% endif %}
```

**Reply by shifuma** [reply]
Oh wow, I didn't realise you could pass Ids to `myMatrixField` like that! Perfect. Thank you as always.

**Reply by brandonkelly** [reply]
Glad that works for you! I just tweaked the code a little to be defensive about when `nestedIds` comes back as an empty array, in which case you probably don’t want to run the `entries` query at all.
